serve_command.go

```diff
@@ -164,11 +161,8 @@ var serveCommand *cli.Command = &cli.Command{
 			}
 		}()
 
-		stop := make(chan os.Signal, 1)
-		signal.Notify(stop, syscall.SIGINT, syscall.SIGTERM)
-
 		// Wait for interrupt signal to gracefully shutdown the server.
-		<-stop
+		<-ctx.Done()
```

Comment:
the ctx is cancellation aware now due to the `NotifyContext` in `main.go`:

<img width="1016" height="125" alt="Image" src="https://github.com/user-attachments/assets/028d1ae9-4357-4b55-82c4-ccd6d75268db" />